### PR TITLE
ATO-441: Multiple relying parties

### PR DESCRIPTION
**Pull request by ethanmills**
## What?

Allow a single instance of the stub to act as multiple relying parties

## Why?

This means that we do not need to create a new stub instance for every RP configuration.

## Related PRs
https://github.com/govuk-one-login/ipv-stubs-common-infra/pull/185

### CODE REVIEW

**Comment by ethanmills** [comment]
This is going to cause some unavailability on the stubs while the new config goes in. I'm going to be watching the deployment of this with values ready to paste in to minimise this.

File: template.yaml [hunk 1]
```diff
@@ -1389,6 +1320,29 @@ Resources:
...
+        Statement:
+          - Effect: Allow
+            Action:
+              - secretsmanager:GetSecretValue
+            Resource: !Ref RpConfigurationSecret
+          - Effect: Allow
+            Action: kms:Decrypt
+            Resource: !GetAtt KmsKey.Arn
+
+  # BEGIN remove
```
Comment (ethanmills) [inline template.yaml:1345]: I'll remove these in a followup, just leaving them here for now to make rollback easier

Comment (CarlyG55) [inline template.yaml:1345]: Just to check, I assume you meant to leave this in? is it for another pr?

File: README.MD [hunk 2]
```diff
@@ -33,31 +40,45 @@ deploy to build, then if successful promote to staging, then if successful promo
...
-your new client in PEM format. Search the file for this string, and you'll be able to find the private key in PKCS8 format.
-You can check you have the right format as it will start with `-----BEGIN PRIVATE KEY-----`. You can then update the secret to be this value.
+your new client in PEM format. Search the file for this string, and you'll be able to find the private key in PKCS8
+format.
+You can check you have the right format as it will start with `-----BEGIN PRIVATE KEY-----`.
+
+
+## Deploying a new copy of the stub
+
+To deploy a new copy of the stub you should create new copies of the resources in the template (except the lambda authorizer,
```
Comment (CarlyG55) [inline README.MD:73]: Maybe add task role in these brackets?

File: template.yaml (hunk 1 above)
Comment (CarlyG55) [reply template.yaml:1345]: Completely missed your comment above 🤦‍♀️ sorry

File: template.yaml [hunk 3]
```diff
@@ -58,50 +58,40 @@ Globals:
 Mappings:
   EnvironmentConfiguration:
     dev:
-      opBaseUrl: https://oidc.sandpit.account.gov.uk/
+      defaultClientId: KxpWnTPIMK3GIuLY6YXHuRl7nPYvDOgB
       stubUrl: https://rp-dev.build.stubs.account.gov.uk
+      docappDefaultClientId: OyjIjatiwdnMtP4eX2gPpFhnLfN2kO9f
```
Comment (CarlyG55) [inline template.yaml:63]: This will need changing

**Review by CarlyG55 — APPROVED** [review]

**Pull request merged**